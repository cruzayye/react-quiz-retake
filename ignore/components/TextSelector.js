import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ text, onChange }) {
  return (
    <input name="text" value={text} onChange={onChange} />
  );
}

TextSelector.propTypes = {
  text: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default TextSelector;
