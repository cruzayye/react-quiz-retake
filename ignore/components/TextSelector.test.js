import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelctor', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<TextSelector text="ryan" onChange={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can change text', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<TextSelector text="ryan" onChange={onChange} />);

    expect(wrapper.find('input').simulate('change', {
      target: {
        name: 'text',
        value: 'myArtist'
      }
    }));
    expect(wrapper.state().text).toEqual('myArtist');
    setTimeout(() => {
    }, 500);
  });
});
