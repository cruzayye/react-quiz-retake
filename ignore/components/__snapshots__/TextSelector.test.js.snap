// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelctor matches snapshot 1`] = `
<input
  name="text"
  onChange={[MockFunction]}
  value="ryan"
/>
`;
